code/frameworks/nextjs/src/utils.ts

```diff
@@ -17,46 +16,13 @@ export const configureRuntimeNextjsVersionResolution = (baseConfig: WebpackConfi
 
 export const getNextjsVersion = (): string => require(scopedResolve('next/package.json')).version;
 
-const findNextConfigFile = async (configDir: string) => {
-  const supportedExtensions = ['mjs', 'js'];
-  return supportedExtensions.reduce<Promise<undefined | string>>(
-    async (acc, ext: string | undefined) => {
-      const resolved = await acc;
-      if (!resolved) {
-        acc = findUp(`next.config.${ext}`, { cwd: configDir });
-      }
-
-      return acc;
-    },
-    Promise.resolve(undefined)
-  );
-};
-
 export const resolveNextConfig = async ({
-  baseConfig = {},
   nextConfigPath,
-  configDir,
 }: {
-  baseConfig?: WebpackConfig;
   nextConfigPath?: string;
-  configDir: string;
 }): Promise<NextConfig> => {
-  const nextConfigFile = nextConfigPath || (await findNextConfigFile(configDir));
-
-  if (!nextConfigFile || (await pathExists(nextConfigFile)) === false) {
-    return {};
-  }
-
-  const nextConfigExport = await import(pathToFileURL(nextConfigFile).href);
-
-  const nextConfig =
-    typeof nextConfigExport === 'function'
-      ? nextConfigExport(PHASE_DEVELOPMENT_SERVER, {
-          defaultConfig: baseConfig,
-        })
-      : nextConfigExport;
-
-  return nextConfig.default || nextConfig;
+  const dir = nextConfigPath ? path.dirname(nextConfigPath) : getProjectRoot();
+  return loadConfig(PHASE_DEVELOPMENT_SERVER, dir, undefined);
```

Comment:
Can you explain how this let you to remove all the code above?

Reply:
For sure!

We don't have to look up or resolve the next config by ourselves since this is encapsulated in `loadConfig`.

Reply:
That is a great improvement then!